### ISSUE TITLE
Requesting clarity on the distinction between HKT<> and Kind<>

### ISSUE THREAD

**Issue by jchitel** [gcanti/fp-ts#954]
## 📖 Documentation

Hello! Right off the bat, I want to mention that I have seen [the HKT documentation](https://gcanti.github.io/fp-ts/recipes/HKT.html) and [this issue](https://github.com/gcanti/fp-ts/issues/882) and both were very helpful for me.

However, I just want to make sure that I understand correctly, and I believe that the output of this issue should be included in the documentation so that it makes sense for other users.

I understand that `HKT<>` and `Kind<>` and their higher-kinded friends are both abstractions over higher-kinded types. Both have the same type signature, but while `HKT<>` is based off of phantom types, `Kind<>` is based off of the `URItoKind<>` dictionaries.

What I do not understand is when to use which one.

In [this issue](https://github.com/gcanti/fp-ts/issues/882), you mention this in response to @amaurymartiny's suggestion that `HKT` and `Kind` are interchangeable (by the way, I am aware that `Kind` used to be called `Type`, and I am replacing mentions of `Type` with `Kind`):

> AFAIK that's because you also augmented `Observable` and `Promise` with `_URI` and `_A`. If you don't do that, you must use `Kind`

This leads me to believe that `HKT` is intended to be used in custom types that define a `_URI` field, and `Kind` is intended to be used in built-in types that do not already define a `_URI` field.

I also realize that each higher-kinded `HKT` type extends the previous one, so `HKT2` extends `HKT` and so on. Is this done so that higher-kinded types can be used in places where a lower-kind is expected?

Basically, I want to know if I'm on the right track with this and what I'm missing. I'd like some concrete use cases describing when you should use `HKT` vs `Kind` and what the technical difference is between the two.

Thanks in advance!

**Comment by gcanti** [comment]
> This leads me to believe that HKT is intended to be used in custom types that define a _URI field, and Kind is intended to be used in built-in types that do not already define a _URI field.

@jchitel no, that's an old encoding (fp-ts v1), with v2 you don't need those fields.

I'm using `HKT` to implement a function

```ts
// Foldable.ts

export function intercalate<M, F>(M: Monoid<M>, F: Foldable<F>): (sep: M, fm: HKT<F, M>) => M
```

`Kind` to handle `* -> *` type constructors (for the same function)

```ts
export function intercalate<M, F extends URIS>(M: Monoid<M>, F: Foldable1<F>): (sep: M, fm: Kind<F, M>) => M
export function intercalate<M, F>(M: Monoid<M>, F: Foldable<F>): (sep: M, fm: HKT<F, M>) => M
```

`Kind2` to handle `* -> * -> *` type constructors (for the same function)

```ts
export function intercalate<M, F extends URIS2>(M: Monoid<M>, F: Foldable2<F>): <E>(sep: M, fm: Kind2<F, E, M>) => M
export function intercalate<M, F extends URIS>(M: Monoid<M>, F: Foldable1<F>): (sep: M, fm: Kind<F, M>) => M
export function intercalate<M, F>(M: Monoid<M>, F: Foldable<F>): (sep: M, fm: HKT<F, M>) => M
```

etc...
